pywbem/_logging.py

```diff
@@ -14,404 +14,341 @@
+    conn = WBEMConnection()  # logs WBEMConnection api calls and responses
+
+* Configure a single WBEMConnection for output of summary information::
+
+    conn = WBEMConnection()
+    WBEMConnection.configure_logger('all', dest='file', log_filname='xxx.log',
+                                        connection=conn)
+
+* Configure logging to log only http requests and responses including all
+  information in the requests and responses to a file::
+
+    WBEMConnection.configure_logger('http', 'file', log_filename='mylog.log')
+    conn = WBEMConnection(..., enable_log='all')
+
+Configure pywbem logging using Python logging methods.
+
+* Configure a single pywbem connection with standard Python logger methods by
+  defining the root logger with basicConfig::
+
+    # Set basic logging configuration (i.e root level logger)
+    import logging
+    logging.basicConfig(filename='example.log',level=logging.DEBUG)
+    conn = WBEMConnection(...)
+
+    # Define the detail_level and WBEMConnection object to activate.
+    WBEMConnection.configure_logger('all', detail_level='all', connection=conn)
+
+
+* Configure the api logger  with a RotatingFileHandler for all subsequent
+  WBEMConnections using Python methods to configure the loggers so all `pywbem`
+  loggers are configured::
+
+    from logging.handlers import RotatingFileHandler
+    # Define logger for pywbem name
+    logger = logging.getLogger('pywbem')  # logger for pywbem.http and pywbem.api
+    handler = logging.RotatingFileHandler("my_log.log", maxBytes=2000,
+                                          backupCount=10)
+    handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(message)s')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+    # configure detail_level and set connection so all WBEMConnection objects
+    # are activated.
+    WBEMConnection.configure_logger('api', detail_level='summary',
+                                    connection=True)
+
+    # All of the following connections do api call and response logging with
+    # summary information.
+    WBEMConnection.WBEMConnection(...)
+    WBEMConnection.WBEMConnection(...)
+
+Configure and activate loggers using :func:`configure_loggers_from_string`.
+
+* Log everything to stderr and output all of the information using the
+  pywbem method that configures the loggers from a string::
+
+    configure_loggers_from_string('all=stderr:summary', connection=True)
+    conn = WBEMConnection()
+
+* Log api calls and responses summary information to stderr::
+
+    configure_loggers_from_string('api=stderr:summary,http=summary',
+                               connection=True)
+    conn = WBEMConnection(...)
 
-import logging
-import six
+The logger names themselves are uniquely identified with an id string for
+each WBEMconnection (:attr:`~pywbem.WBEMConnection.conn_id`) so that different
+loggers can be configured for each :class:~pywbem.WBEMConnection
```

Comment:
It is not because of the connection id that different loggers can be configured for each connection, but because there are two loggers implemented :-) I think you wanted to say something different, here?

Reply:
Actually both.  There are two logger names at the level of the configure.  However, each call to creat a log entries has a third level added, the conn_id so that the logs are created with the name like
pywbem.api-1-2345.

Reply:
Made a couple of text changes to clarify than